### DISCUSSION TITLE
[Next.js] Noticed painfully slow compile times in development

### DISCUSSION THREAD

**Discussion by iyifr** [chakra-ui/panda#2179] [General]
I noticed a weird pattern of slow compile times when using PandaCSS with Next.js in development. With fast refresh, it can take about 20+ seconds to extract styles and show the new content on the screen for the smallest changes which I find very bad for DX.

![image](https://github.com/chakra-ui/panda/assets/105992164/851944b0-23ad-4217-b912-e2a97ad67f0d)
I don't know if this should be an issue or not but can anyone help with this?

**Comment by astahmer** [comment]
1. what version of Panda are you using ? we made huge improvements between 0.22 and 0.27x, upgrading to latest (currently 0.30.2) is probably safe
2. [can you reproduce the issue with the CLI ?](https://panda-css.com/docs/guides/debugging#the-postcss-plugin-sometimes-seems-slow-or-runs-too-frequently)
3. when using [`panda debug`](https://panda-css.com/docs/guides/debugging#panda-debug), does it seem like there's a lot of files being processed that do not contain Panda styling ? (the last line will look like `info [cli] Found 100/159 files using Panda`)

if so, that means your [`include`](https://panda-css.com/docs/references/config#include) is most likely not specific enough or as an alternative you could add some file globs in your [`exclude`](https://panda-css.com/docs/references/config#exclude)

[if you can still reproduce the performance issue, can you send the resulting .cpuprofile after following these steps ?](https://panda-css.com/docs/guides/debugging#performance-profiling)

**Comment by iyifr** [comment]
I figured out the issue was with a Radix colors preset I was using which had heavy objects being parsed on each render. I've resolved the issue by only using the colors I need. The compilation is better now having switched 0.30.2
